extension_helpers/tests/test_setup_helpers.py

```diff
@@ -10,6 +10,27 @@
 from .._setup_helpers import get_compiler, get_extensions
 from . import cleanup_import, run_setup
 
+if sys.version_info >= (3, 11):
+    from contextlib import chdir
+else:
+    # vendored from Python 3.11b1
```

Comment:
We should probably include the Python license if this is copied from the CPython source

Reply:
Sure. Where should I put it ?

Reply:
The cleanest might be to actually put this context manager in a separate Python file and include the license inside it and then when we no longer need to support it we can remove file

Reply:
done !